---
title: "Defined synthetic microbial communities colonize and benefit field-grown sorghum"
author: "Citlali Fonseca-Garcia"
date: '2023-05-31'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown  
This script is a modified version of the Rmd file included in the manuscript "Drought shifts sorghum root metabolite and microbiome profiles and enriches the stress response factor pipecolic acid" designed by Daniel Caddel.  

This document includes analyses of the 16S rRNA amplicon sequencing of root and rhizosphere from the dSynCom in vitro and in planta experiments.

The following scripts should run as is assuming you have downloaded all data and markdown files to the same directory. 

## load packages
```{r}
library(dada2);packageVersion("dada2")
library(phyloseq); packageVersion("phyloseq")
library(ape);packageVersion("ape")
library(data.table);packageVersion("data.table")
library(ggplot2); packageVersion("ggplot2")
library(reshape2); packageVersion("reshape2")
library(dplyr); packageVersion("dplyr")
library(RColorBrewer); packageVersion("RColorBrewer")
library(beepr); packageVersion("beepr")
library(scales); packageVersion("scales")
library(grid); packageVersion("grid")
library(ggalt); packageVersion("ggalt")
library(mctoolsr); packageVersion("mctoolsr")
library(vegan); packageVersion("vegan")
library(agricolae);packageVersion("agricolae")
library(patchwork);packageVersion("patchwork")
library(ggfortify);packageVersion("ggfortify")
library(stringr);packageVersion("stringr")
library(PMCMRplus);packageVersion("PMCMRplusr")     
require(multcompView);packageVersion("multcompView")
theme_set(theme_bw()) #Define a default theme for ggplot graphics.

#setwd("/set/working/directory/location/")

```

## import OTU biom file, sample file, and tree file, then build phyloseq object  
update file locations with correct paths
```{r}
biom_file <- "~merged-feature-table_Devin.biom" #update path info
map_file <- "~sample-metadata-final_feature DSynCom_Lab.txt" #update path info
otutax <- import_biom(biom_file, parseFunction = parse_taxonomy_default)
sam <- import_qiime_sample_data(map_file)
all <- merge_phyloseq(otutax, sam)
all

colnames(tax_table(all)) <- c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")
colnames(tax_table(all))
```

## Remove mitochondrial and chloroplast reads from a phyloseq object (Illumina - greengenes).  
A full list of related functions are in 'functions_for16SData.Rmd'.  
WARNING: only use on taxonomies generated by greengenes, otherwise pattern matching will not find mito/chloro ASVs.
This chunk has been modified from code originally written by Alex Styer.
```{r}

remove_mitoAndChloro <- function(x){#where x is a merged phyloseq object
  df <- as.data.frame(tax_table(x))
  setDT(df, keep.rownames = TRUE)
  colnames(df)[1] <- "ASV"
  allASVs <- df[,1]
  mito <- filter(df, df$Family == "f__Mitochondria")
  chloro <- filter(df, df$Order == "o__Chloroplast")
  contam <- as.vector(rbind(mito, chloro)$ASV)
  ASVs_toKeep <- as.vector(subset(allASVs, !(ASV %in% contam))$ASV)
  num_mito <- nrow(mito)
  num_chloro <- nrow(chloro)
  cat(sprintf("%i and %s ASVs were removed as pesky mitochondria or chloroplast, respectively.\nKhaleesi frees these bacterial slaves to the wild binary yonder.", num_mito, num_chloro))
  x <- prune_taxa(ASVs_toKeep, x)
}
```

## Clean up taxonomy
Takes a psmelted phyloseq object and cleans up greengenes-format taxonomies.  
i.e. removes the "k__" in k__Bacteria.  
Also takes NA values and unassigned taxa and renames their taxonomy as "Unknown".
This chunk has been modified from code originally written by Alex Styer.
```{r}
cleanup_taxonomy <- function(x){ #to be used on dfs made from melted phyloseq objects
  x$Domain <- as.character(lapply(x$Domain, sub, pattern = "d__", replacement = ""))
  x$Phylum <- as.character(lapply(x$Phylum, sub, pattern = "p__", replacement = ""))
  x$Class <- as.character(lapply(x$Class, sub, pattern = "c__", replacement = ""))
  x$Order <- as.character(lapply(x$Order, sub, pattern = "o__", replacement = ""))
  x$Family <- as.character(lapply(x$Family, sub, pattern = "f__", replacement = ""))
  x$Genus <- as.character(lapply(x$Genus, sub, pattern = "g__", replacement = ""))
  x$Species <- as.character(lapply(x$Species, sub, pattern = "s__", replacement = ""))
  x$Domain <- ifelse(x$Domain == "", "Unknown", as.character(x$Domain))
  x$Phylum <- ifelse(x$Phylum == "", "Unknown", as.character(x$Phylum))
  x$Class <- ifelse(x$Class == "", "Unknown", as.character(x$Class))
  x$Order <- ifelse(x$Order == "", "Unknown", as.character(x$Order))
  x$Family <- ifelse(x$Family == "", "Unknown", as.character(x$Family))
  x$Genus <- ifelse(x$Genus == "", "Unknown", as.character(x$Genus))
  x$Species <- ifelse(x$Species == "", "Unknown", as.character(x$Species))
  x$Domain <- ifelse(is.na(x$Domain) == TRUE , "Unknown", as.character(x$Domain))
  x$Phylum <- ifelse(is.na(x$Phylum) == TRUE , "Unknown", as.character(x$Phylum))
  x$Class <- ifelse(is.na(x$Class) == TRUE , "Unknown", as.character(x$Class))
  x$Order <- ifelse(is.na(x$Order) == TRUE , "Unknown", as.character(x$Order))
  x$Family <- ifelse(is.na(x$Family) == TRUE , "Unknown", as.character(x$Family))
  x$Genus <- ifelse(is.na(x$Genus) == TRUE , "Unknown", as.character(x$Genus))
  x$Species <- ifelse(is.na(x$Species) == TRUE , "Unknown", as.character(x$Species))
  x <- x
}
```

## Additional Phyloseq tools  

A quantitative/informative tool for deciding which number of top ASVs to keep for stacked bar plots.  
Takes a phyloseq object and plots each ASV's relative abundance against its rank from 1-N (1 being most abundant, N least).  
Also prints to console summary statistics decribing the number of ASVs that represent a given fraction of readcounts.
This chunk has been modified from code originally written by Alex Styer.
```{r}
plot_relAbund_curve <- function(x){
  df <- as.data.frame(rowSums(otu_table(x)))
  setDT(df, keep.rownames = TRUE)
  colnames(df) <- c("ASV", "ASV_sum")
  all_sum <- sum(df$ASV_sum)
  df$ASV_relAbund <- df$ASV_sum/all_sum
  df <- df[order(-ASV_relAbund)]
  df$ASV_rank <- 1:nrow(df)
  
  #a function to get percentiles
  get_percentile <- function(y, n){#n is a double between 0-1; the output will tell you the smallest number of ASVs that account for that perecnt relative abundance in your dataset.
    row = 1
    num = 0
    while(num<=n){
      num = sum(y$ASV_relAbund[1:row])
      row = row + 1
    }
    y$ASV_rank[row-1]
  }
  
  #a function to print percentiles to console
  print_percentiles <- function(a,b,c,d,e,f,g,h,i){
    cat(sprintf("the top %i ASVs = ~10%% of all reads
the top %i ASVs = ~20%% of all reads
the top %i ASVs = ~30%% of all reads
the top %i ASVs = ~40%% of all reads
the top %i ASVs = ~50%% of all reads
the top %i ASVs = ~60%% of all reads
the top %i ASVs = ~70%% of all reads
the top %i ASVs = ~80%% of all reads
the top %i ASVs = ~90%% of all reads"
                ,a,b,c,d,e,f,g,h,i))
  }
  
  #get summary statistics
  mean_relabund <- mean(df$ASV_relAbund)
  per10 <- get_percentile(df, .1)
  per20 <- get_percentile(df, .2)
  per25 <- get_percentile(df, .25)
  per30 <- get_percentile(df, .3)
  per40 <- get_percentile(df, .4)
  per50 <- get_percentile(df, .5)
  per60 <- get_percentile(df, .6)
  per70 <- get_percentile(df, .7)
  per75 <- get_percentile(df, .75)
  per80 <- get_percentile(df, .8)
  per90 <- get_percentile(df, .9)
  
  #set plot aesthetics and add-ons
  x_title <- scale_x_continuous(name = "ASV Rank", breaks = seq(0, max(df$ASV_rank), by = 10))
  y_title <- scale_y_continuous(name = "ASV Relative Abundance")
  vline25 <- geom_vline(xintercept = per25, color="#444546")
  vline25_text <- geom_text(aes(x=per25, y=(max(df$ASV_relAbund)/2), label=(sprintf("top %i ASVs = ~25%% reads", per25))), vjust=-1, angle=90, size=10/.pt, color="#444546", check_overlap = TRUE)
  vline50 <- geom_vline(xintercept = per50, color="#444546")
  vline50_text <- geom_text(aes(x=per50, y=(max(df$ASV_relAbund)/2), label=(sprintf("top %i ASVs = ~50%% reads", per50))), vjust=-1, angle=90, size=10/.pt, color="#444546", check_overlap = TRUE)
  vline75 <- geom_vline(xintercept = per75, color="#444546")
  vline75_text <- geom_text(aes(x=per75, y=(max(df$ASV_relAbund)/2), label=(sprintf("top %i ASVs = ~75%% reads", per75))), vjust=-1, angle=90, size=10/.pt,color="#444546", check_overlap = TRUE)
  hline_mean <- geom_hline(yintercept = mean_relabund, color="#a04344")
  hline_mean_text <- geom_text(aes(x=(max(df$ASV_rank)/2), y=mean_relabund, label=(sprintf("mean relative abundance = %f", mean_relabund))), vjust = -1, size=10/.pt,color="#a04344", check_overlap = TRUE)

  #plot ASV rank by relative abundance
  if (nrow(df)<=300) {
    print_percentiles(per10, per20, per30, per40, per50, per60, per70, per80, per90)
    ggplot(df, aes(x=ASV_rank, y=ASV_relAbund)) + 
      geom_point() +
      list(x_title, y_title, vline25, vline25_text, vline50, vline50_text, vline75, vline75_text, hline_mean, hline_mean_text)
  }else{
    if (nrow(df)>300 && per75<=300) {
      df <- df[1:300]
      cat("Trimming data to top 300 ASVs for plot readability\nSummary statistics represent all input ASVs\n")
      print_percentiles(per10, per20, per30, per40, per50, per60, per70, per80, per90)
      ggplot(df, aes(x=ASV_rank, y=ASV_relAbund)) + 
        geom_point() +
        list(x_title, y_title, vline25, vline25_text, vline50, vline50_text, vline75, vline75_text, hline_mean, hline_mean_text)
    }else{
      if (nrow(df)>300 && per50<=300 && per75>300) {
        df <- df[1:300]
        cat("Trimming data to top 300 ASVs for plot readability\nSummary statistics represent all input ASVs\n")
        print_percentiles(per10, per20, per30, per40, per50, per60, per70, per80, per90)
        ggplot(df, aes(x=ASV_rank, y=ASV_relAbund)) + 
          geom_point() +
          list(x_title, y_title, vline25, vline25_text, vline50, vline50_text, hline_mean, hline_mean_text)
      }else{
        if (nrow(df)>300 && per25<=300 && per50>300) {
          df <- df[1:300]
          cat("Trimming data to top 300 ASVs for plot readability\nSummary statistics represent all input ASVs\n")
          print_percentiles(per10, per20, per30, per40, per50, per60, per70, per80, per90)
          ggplot(df, aes(x=ASV_rank, y=ASV_relAbund)) + 
            geom_point() +
            list(x_title, y_title, vline25, vline25_text, hline_mean, hline_mean_text)
        }else{
          df <- df[1:300]
          cat("Trimming data to top 300 ASVs for plot readability\nSummary statistics represent all input ASVs\n")
          print_percentiles(per10, per20, per30, per40, per50, per60, per70, per80, per90)
          ggplot(df, aes(x=ASV_rank, y=ASV_relAbund)) + 
            geom_point() +
            list(x_title, y_title, hline_mean, hline_mean_text)
        }
      }
    }
  } 
}
```

Now run the cleanup functions to remove any remaining mitochondria and chloroplast sequences.  
Also melt phyloseq object into a dataframe and transform sample counts for further analyses.  
```{r}
all.clean <- remove_mitoAndChloro(all)
all.clean.df <- psmelt (all.clean)
all.clean.df <- cleanup_taxonomy(all.clean.df)
#transform data
all.trans <- transform_sample_counts(all.clean, function(OTU) 100*OTU/sum(OTU))
#merge groups within transformed data
all.trans.merge <- merge_samples(all.trans, "Group")
all.trans.merge <- transform_sample_counts(all.trans.merge, function(OTU) 100*OTU/sum(OTU))
all.tm.df<- psmelt(all.trans.merge)
all.tm.df <- cleanup_taxonomy(all.tm.df) #NOTE: some columns were messed up during merge_samples

beep(4)
```

## Save the session so that it can be loaded later without having to rerun this step.
This will save time compared to reruning the code above.
```{r eval=FALSE, include=TRUE}
save.image("~phyloseq_illuminaASV.rdata")
#close R, re-open R
load("~phyloseq_illuminaASV.rdata")
```

Check the total reads per sample, and the distribution of reads.
```{r}
readsumsdf = data.frame(nreads = sort(taxa_sums(all.clean), TRUE), sorted = 1:ntaxa(all.clean), 
                        type = "OTUs")
readsumsdf = rbind(readsumsdf, data.frame(nreads = sort(sample_sums(all.clean), 
                                                        TRUE), sorted = 1:nsamples(all), type = "Samples"))
title = "Total number of reads"
p = ggplot(readsumsdf, aes(x = sorted, y = nreads)) + geom_bar(stat = "identity")
p + ggtitle(title) + scale_y_log10() + facet_wrap(~type, 1, scales = "free")

plot_relAbund_curve(all.clean) # this function is from 'functions_for16SData.Rmd'
```

## Alpha Diversity   
Note: this analysis is performed using rarefied data, which differs from the transformation method performed on the data above.
```{r}
set.seed(42)
all_rare<-rarefy_even_depth(all.clean, rngseed = TRUE)
#sanity check
par(mfrow = c(1, 2))
title = "Sum of reads for each sample, all_rare"
plot(sort(sample_sums(all_rare), TRUE), type = "h", main = title, ylab = "reads") 
#, ylim = c(0, 1000))

richness = estimate_richness(all_rare,measures = "Shannon")
richness
```

ANOVA of alpha-diversity and Tukey-HSD posthoc test
```{r}
temp.aov <- sample_data(all.clean)
temp.aov <- temp.aov[,c(1:8)]
temp.aov$ST <- paste0(temp.aov$Treatment, temp.aov$Water.Media, temp.aov$Sampletype.Week) #add column to see if sample diversity changes with timepoint
data.aov <- cbind(temp.aov, richness)
data.aov

#Kruskal-Wallis test, Dunn's Test of Multiple Comparisons
kruskal.test(data_new$Shannon, data_new$Group)
dunn <-kwAllPairsDunnTest(Shannon ~ Group, data = data_new, p.adjust="none")
  out.p <- get.pvalues(dunn)
  out.mcV <- multcompLetters(out.p, threshold=0.05)
  out.mcV

```

Plot of Shannon Diversity Index, Supplemental figure 4A
```{r}

# Duplicate data
data_new <- data.aov                             
data_new$Group <- factor(data_new$Group, c("SynCom.SSM.1","SynCom.SSM.2","SynCom.SSM.3","SynCom.SSM.4","SynCom.SSM.5","SynCom.SSM.6",
"SynCom.SSM.7","SynCom.SSM.8","SynCom.SSMEx.1","SynCom.SSMEx.2","SynCom.SSMEx.3","SynCom.SSMEx.4","SynCom.SSMEx.5","SynCom.SSMEx.6","SynCom.SSMEx.7","SynCom.SSMEx.8","Mock.Water.Rh","SynCom.Water.Rh","Mock.Drought.Rh","SynCom.Drought.Rh","Mock.Water.R","SynCom.Water.R","Mock.Drought.R","SynCom.Drought.R")) # Reorder factor levels

p<-ggplot(data_new, aes(x=Group, y=Shannon, fill=Group)) + xlab("") + ylab("Shannon index")+
  geom_boxplot(position=position_dodge(1)) + theme(axis.text.x = element_text(angle = 90, hjust=1, face="bold"), legend.position="none")
p

# Change colors
p2<- p+scale_fill_manual(values=c("#F7F4F9","#E7E1EF","#D4B9DA","#C994C7","#DF65B0","#E7298A", "#CE1256", "#91003F","#F7F4F9","#E7E1EF","#D4B9DA","#C994C7","#DF65B0","#E7298A", "#CE1256", "#91003F","lightskyblue1","#7997a1", "lightgoldenrod1","#cb9e59","lightskyblue3","#4d5b6a", "lightgoldenrod3","#cdbe70", "#4d5b6a","#686a47","#cb9e59","#7997a1","peachpuff3","olivedrab4","#4d5b6a","#686a47","#cb9e59","#7997a1","peachpuff3","olivedrab4"))

p2
ggsave("~Alpha-diversity.png", plot=p2, width=9, height=7, dpi=600)

```

## Beta Diversity
Beta diversity (PCoA) of sorghum root, rhizosphere, and soil microbiomes at TP8 and 24 hours after rewatering in well-watered control or drought plots. Figure 1D
```{r}

#Plot MDS
all.trans@sam_data$Treatment <- factor(all.trans@sam_data$Treatment, c("SSM","SSMEx","Mock","SynCom")) # Reorder factor levels
all.trans@sam_data$SampleWater <- factor(all.trans@sam_data$SampleWater, c("Week1","Week2","Week3","Week4","Week5",
"Week6","Week7","Week8","Watered_Rh","Watered_R","Drought_Rh","Drought_R")) # Reorder 

ord.mds.bray <- ordinate(all.trans, method="MDS", distance="bray")
p1d <- phyloseq::plot_ordination(all.trans, ord.mds.bray, color="SampleWater", shape="Treatment", title="Bray-Curtis distance")+
          geom_vline(xintercept = 0,linetype="longdash",color="gray")+
          geom_hline(yintercept = 0,linetype="longdash",color="gray")+ scale_shape_manual(values = c( 15, 19,18,17)) +  geom_point( size = 8) +
          scale_color_manual(values=c("#F7F4F9","#E7E1EF","#D4B9DA","#C994C7","#DF65B0","#E7298A", "#CE1256", "#91003F","#7997a1","#4d5b6a","#cb9e59","#cdbe70","#686a47","peachpuff3"))+
          theme(axis.text.x=element_text(size=14,color="black",angle=0), 
                axis.text.y=element_text(size=14,color="black"), 
                axis.title=element_text(size=14,face="bold"), 
                text=element_text(size=14, face="bold"),
                plot.title=element_text(size=15,hjust= 0.5, vjust=2),
                legend.title=element_text(size=13), 
                legend.text=element_text(size=14),
                legend.key.size = unit(3, 'mm'))
p1d

ggsave("~Beta-diversity.png", plot = p1d, width=11, height=9, dpi=600)
```

Adonis test and pairwise permanova posthoc test
```{r}
ps.prop_dist <- distance(all.trans, method = "bray")
ps.prop_data <- data.frame(sample_data(all.trans))
summary(ps.prop_data)

#Global PERMANOVA
test <- adonis(ps.prop_dist ~ Treatment + SampletypeWeek + WaterMedia + Group + SampleWater, data = ps.prop_data)
View(test)
write.table(test[["aov.tab"]], file = "~aov.tab_Global.csv", append = FALSE, quote = TRUE, sep = ",", row.names = TRUE, col.names = TRUE)

#in vitro PERMANOVA, load metadata named: sample-metadata-final_feature DSynCom_plates
test <- adonis(ps.prop_dist ~ Treatment + SampletypeWeek + WaterMedia + Group + SampleWater, data = ps.prop_data)
View(test)
write.table(test[["aov.tab"]], file = "~aov.tab_invitro.csv", append = FALSE, quote = TRUE, sep = ",", row.names = TRUE, col.names = TRUE)

#in planta PERMANOVA, load metadata named: sample-metadata-final_feature DSynCom_planta
test <- adonis(ps.prop_dist ~ Treatment + SampletypeWeek + WaterMedia + Group + SampleWater, data = ps.prop_data)
View(test)
write.table(test[["aov.tab"]], file = "~aov.tab_inplanta.csv", append = FALSE, quote = TRUE, sep = ",", row.names = TRUE, col.names = TRUE)


```

Heat map of the abundance of each dSynCom member across in vitro and in planta samples . Figure 1C 
```{r}
#plot a heat map of plates and in planta experiments 
Data <- read.table("~MembersASV-Plates-Full_namedEdited.txt", header=TRUE) #update path info
Data[Data == 0] <- 1

#make a heatmap without details
pheatmap(Data, cluster_cols = FALSE, cluster_rows = FALSE, scale = "column")

# Generate annotations for rows and columns
annotation_col = data.frame(
  Sample = factor (rep(c("Week1", "Week2","Week3","Week4","Week5","Week6", "Week7", "Week8", 
                         "Week1", "Week2","Week3","Week4","Week5","Week6", "Week7", "Week8",
                         "Root", "Rhizosphere","Root", "Rhizosphere"), c(3,4,4,3,4,4,4,4,3,2,3,4,4,4,4,4,19,20,19,20))),
  Media = factor(rep(c("SSM", "SSMEx", "Watered", "Drought"), c(30, 28, 39, 39))))
 
rownames(annotation_col) = colnames(all.tm.df)

##### Reordered by Gram + or - the plot
annotation_row = data.frame(
  Genus = factor(rep(c("Kribbella",
                        "Rhodococcus",
                        "Streptomyces",
                        "Bacillus",
                        "Paenibacillus",
                        "Chryseobacterium",
                        "Flavobacterium",
                        "Rhizobium",
                        "Sphingobium",
                        "Acinetobacter",
                        "Paraburkholderia",
                        "Enterobacter",
                        "Lysobacter",
                        "Pseudomonas"), c(1, 2, 3, 6, 3, 3, 1, 1, 1, 1, 2, 1,1, 7))))


rownames(annotation_row) = rownames(all.tm.df)
# Specify colors
ann_colors = list(
    Sample = c(Week1="#F7F4F9", Week2="#E7E1EF",Week3="#D4B9DA",Week4="#C994C7",Week5="#DF65B0",Week6="#E7298A", Week7="#CE1256", Week8="#91003F", 
             Root="#cd3700", Rhizosphere="#721e00"),
  Media = c(SSM = "#ffe4c4", SSMEx = "#800080", Watered = "#0000ff", Drought ="#ffa500"),
  Genus = c (Kribbella="#B3B86A",
             Rhodococcus="#C55C73",
             Streptomyces="#BF7A8F",
             Bacillus="#5C89A9",
             Paenibacillus="#E6AA59",
             Chryseobacterium ="#41817B",
             Flavobacterium="#6BA56F",
             Rhizobium="#B64759",
             Sphingobium="#b76981",
             Acinetobacter="#599EC4",
             Paraburkholderia="#D65F3C",
             Enterobacter="#3E8B67",
             Lysobacter="#dce659",
             Pseudomonas="#B14446"))
 
p1b <- pheatmap(log2(Data), scale = "none", cluster_cols = FALSE, cluster_rows = FALSE,annotation_col = annotation_col, fontsize = 15.5,
                show_rownames = T, show_colnames = F, annotation_row = annotation_row,annotation_colors = ann_colors, gaps_row = c(6, 15, 19, 21), gaps_col = c(30, 58, 58, 97), angle_col = "90")

ggsave("~HeatmapStrainLabLog2.png", plot = p1b, width=22, height=10, dpi=600)


```

Heat map of the abundance of each dSynCom member without sorghum-exudate solubilizers across in vitro and in planta samples . Figure 3
```{r}
##without sorghum-exudate solubilizers isolates only, both lab based experiments
Data <- read.table("~MembersASV-ColemanStrainsNamesEdited.txt", header=TRUE) #update path info
Data[Data == 0] <- 1

#sorghum-exudate solubilizers isolates experiment data only
Data_F <- Data[,c(59:168)]

#make a heatmap without details
pheatmap(Data_F, cluster_cols = FALSE, cluster_rows = FALSE, scale = "column")

# Generate annotations for rows and columns
annotation_col = data.frame(
  Sample = factor (rep(c("Root","Root", "Rhizosphere","Root", "Rhizosphere"), c(32,19,20,19,20))),
  Irrigation = factor(rep(c("Watered", "Drought", "Watered", "Drought"), c(16, 16, 39, 39))))
  #ExpType = factor(rep(c("No_SH","Complite"), c(32,78))))

rownames(annotation_col) = colnames(all.tm.df)


##### Reordered by Gram + or - the plot
annotation_row = data.frame(
  Genus = factor(rep(c("Kribbella",
                        #"Rhodococcus",
                        "Streptomyces",
                        "Bacillus",
                        "Paenibacillus",
                        "Chryseobacterium",
                        #"Flavobacterium",
                        "Rhizobium",
                        #"Sphingobium",
                        #"Acinetobacter",
                        #"Paraburkholderia",
                        #"Enterobacter",
                        "Lysobacter",
                        "Pseudomonas"), c(1, 3, 6, 3, 2, 1, 1, 3))))
 
rownames(annotation_row) = rownames(all.tm.df)
# Specify colors
ann_colors = list(
  Sample = c(Root="#cd3700", Rhizosphere="#721e00"),
  Irrigation = c(Watered = "#0000ff", Drought ="#ffa500"),
    Genus = c (Kribbella="#B3B86A",
                          #Rhodococcus="#C55C73",
                          Streptomyces="#BF7A8F",
                          Bacillus="#5C89A9",
                          Paenibacillus="#E6AA59",
                          Chryseobacterium ="#41817B",
                          #Flavobacterium="#6BA56F",
                          Rhizobium="#B64759",
                          #Sphingobium="#b76981",
                          #Acinetobacter="#599EC4",
                          #Paraburkholderia="#D65F3C",
                          #Enterobacter="#3E8B67",
                          Lysobacter="#dce659",
                          Pseudomonas="#B14446"))
  
p1b <- pheatmap(log2(Data_F), cluster_cols = FALSE, cluster_rows = FALSE,annotation_col = annotation_col, scale = "column", fontsize = 16,
                annotation_row = annotation_row,annotation_colors = ann_colors, annotation_names_row = TRUE, annotation_names_col = TRUE,
                show_rownames = T, show_colnames = F, gaps_row = c(4, 13, 15, 17),gaps_col = c(16,32,32,71), angle_col = "90")

ggsave("~HeatmapStrainLog2-SES.png", plot = p1b, width=22, height=10, dpi=600)

```
